### PR TITLE
refs #199: Mermaid support

### PR DESCRIPTION
**Pull request by mgdesign**
Added basic Mermaid graph rendering functionality.

### Added 
- Mermaid code blocks are now rendered with a specific class
- Mermaid JS is now included in the pages via CSN

### Changed
- node_modules folders is now mounted inside the container to allow the developer to modify Raneto's code and create a new patch for it
- former code highlight JS functions and CSS classes were slightly renamed to be more readable

### CODE REVIEW

File: custom/themes/spark-playbook/public/js/app.js [hunk 1]
```diff
@@ -44,22 +44,40 @@ $(() => {
...
   }
 
+  function booleanToogle(button) {
+    button.classList.toggle('on');
+  }
+
+  function showDiagramCode(diagram) {
+    const source = diagram.nextSibling;
+    if (source && diagram) {
+      diagram.classList.toggle('hide');
```
Comment (stickgrinder) [inline custom/themes/spark-playbook/public/js/app.js:70]: I'm pretty sure this is correct but can you explain what's going on? It seems you're hiding everything...

File: custom/themes/spark-playbook/scss/_components.code-highlight.scss [hunk 2]
```diff
@@ -117,13 +137,37 @@ code {
   }
 }
 
-.code-copy-shout {
+.code-ui--button--copy {
```
Comment (stickgrinder) [inline custom/themes/spark-playbook/scss/_components.code-highlight.scss:140]: Does this replace the copy button on code blocks too?

**Review by stickgrinder — COMMENTED** [review]
LGTM, just a couple questions.

File: custom/themes/spark-playbook/public/js/app.js (hunk 1 above)
Comment (mgdesign) [reply custom/themes/spark-playbook/public/js/app.js:70]: You're correct in the end: the diagram and its source code are rendered as visible, the first, and hidden, the latter. When you "toggle" the diagram and the source, you are switching these statuses, so they should always be the opposite one of the other. We could reinforce this with a simple check on the first to determine the status of the second element. What do you think?

File: custom/themes/spark-playbook/scss/_components.code-highlight.scss (hunk 2 above)
Comment (mgdesign) [reply custom/themes/spark-playbook/scss/_components.code-highlight.scss:140]: Yes, it affects both of them.

**Pull request merged**